src/Engine/Manager/RenderManager.cpp

```diff
@@ -321,12 +322,22 @@ void RenderManager::Render(World& world, const glm::mat4& translationMatrix, con
 
 }
 
-Component::Animation* RenderManager::CreateAnimation() {
-    return animations.Create();
+void RenderManager::UpdateAnimations(float deltaTime) {
+    // Update all enabled animation controllers.
+    for (Component::AnimationController* animationController : animationControllers.GetAll()) {
+        if (animationController->IsKilled() || !animationController->entity->enabled)
+            continue;
+    
+        animationController->UpdateAnimation(deltaTime);
+    }
+}
+
+Component::AnimationController* RenderManager::CreateAnimation() {
+    return animationControllers.Create();
 }
 
-Component::Animation* RenderManager::CreateAnimation(const Json::Value& node) {
-    Component::Animation* animation = animations.Create();
+Component::AnimationController* RenderManager::CreateAnimation(const Json::Value& node) {
+    Component::AnimationController* animationController = animationControllers.Create();
     
     // Load values from Json node.
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'GetAnimations' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)